### DISCUSSION TITLE
Solved: Trying to upgrade craft 4 to 5

### DISCUSSION THREAD

**Discussion by boldpaul** [craftcms/cms#15521] [General]
Trying to upgrade from craft 4 to 5 now.. Im following the docs (https://craftcms.com/docs/5.x/upgrade.html), here's what ive done so far:
1.Update Craft 4 and plugins to the latest (in Control Panel)
2.Update Craft CMS through Control Panel to the latest available version
3.Update all the remaining plugins through Control Panel to the latest available version
4.Edit my project’s composer.json to require "craftcms/cms": "^5.0.0" and Craft-4-compatible plugins all at once
5.Run ddev composer update

Then, when i run `ddev php craft up`, it runs succssfully:
*** applied m240115_000000_craft5 (time: 5.657s)
28 migrations were applied.
Migrated up successfully.
Checking for pending content migrations ...
No new migrations found. Your system is up-to-date.

 then my web page has the following error:

Warning: require(/var/www/html/bootstrap.php): Failed to open stream: No such file or directory in /var/www/html/web/index.php on line 7

Fatal error: Uncaught Error: Failed opening required '/var/www/html/bootstrap.php' (include_path='.:/usr/share/php') in /var/www/html/web/index.php:7 Stack trace: #0 {main} thrown in /var/www/html/web/index.php on line 7

Here is my composer.json:
```json

{
    "require": {
        "craftcms/ckeditor": "4.1.0",
        "craftcms/cms": "^5.0.0",
        "craftcms/redactor": "4.1.0",
        "doublesecretagency/craft-siteswitcher": "3.0.0",
        "doublesecretagency/craft-viewcount": "2.0.0",
        "mmikkel/retcon": "3.2",
        "mutation/translate": "4.0.2",
        "putyourlightson/craft-entry-count": "4.0.0",
        "topshelfcraft/wordsmith": "5.0.0",
        "verbb/knock-knock": "3.0.1",
        "verbb/super-table": "4.0.2",
        "vlucas/phpdotenv": "^5.6.0",
        "yiisoft/yii2-symfonymailer": "^4.0"
    },
```

**Comment by boldpaul** [comment]
Update:

For some reason, some of the files had the wrong permissions. I set autoload.php and bootstrap.php to 755, now it works!

**Reply by brandonkelly** [reply]
Glad you got it sorted!
